src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -360,7 +360,7 @@ internal BoundExpression BindValueAllowArgList(ExpressionSyntax node, Diagnostic
             out BoundExpression valueBeforeConversion)
         {
             Debug.Assert(this.InParameterDefaultValue);
-            Debug.Assert(this.ContainingMember().Kind == SymbolKind.Method || this.ContainingMember().Kind == SymbolKind.Property);
+            Debug.Assert(this.ContainingMemberOrLambda.Kind == SymbolKind.Method || this.ContainingMemberOrLambda.Kind == SymbolKind.Property);
```

Comment:
Both checks should be performed on the same symbol.

---
In reply to: [98314006](https://github.com/dotnet/roslyn/pull/16789#discussion_r98314006) [](ancestors = 98314006,98313514,98312669)